source/common/network/io_socket_handle_impl.h

```diff
@@ -128,7 +124,11 @@ class IoSocketHandleImpl : public IoSocketHandleBaseImpl {
   // Should only be used by UDP sockets to avoid creating multiple address instances for the same
   // address in each read operation. Only be instantiated if the non-zero address_cache_max_capacity
   // is passed in during the construction.
```

Comment:
Not your change but can you comment that this is currently only in use for QUIC client sockets, where the expected number of addresses is generally 1, and the cache size defaults to 4?

Reply:
Done in https://github.com/envoyproxy/envoy/pull/37902